packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2507,406 @@ describe("Runtime", () => {
+				);
+			});
+
+			it("does not emit error events when signals are processed in order", () => {
+				sendSignals(100);
+				processSubmittedSignals(100);
+
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+						},
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+					],
+					"SignalLost and SignalOutOfOrder telemetry should not be logged when signals are processed in order",
+				);
+			});
+
+			it("logs relative lost signal count in SignalLost telemetry", () => {
+				sendSignals(5);
+
+				dropSignals(1);
+
+				processSubmittedSignals(1);
+
+				// Missing signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+
+				dropSignals(2);
+
+				processSubmittedSignals(1);
+
+				// Missing 3rd and 4th signals should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 2,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+			});
+
+			it("ignores in-flight signals on disconnect/reconnect", () => {
+				sendSignals(4);
+
+				// Disconnect + Reconnect
+				changeConnectionState(containerRuntime, false, mockClientId);
+				changeConnectionState(containerRuntime, true, mockClientId);
+
+				// Temporarily lose two old signals
+				dropSignals(2);
+
+				// Receive old signals sent before disconnect
+				processSubmittedSignals(2);
+
+				// Receive old out of order signals
+				processDroppedSignals(2);
+
+				// No error events should be logged for signals sent before disconnect
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+						{
+							eventName: "ContainerRuntime:SignalLost",
+						},
+					],
+					"SignalOutOfOrder/SignalLost telemetry should not be logged on reconnect",
+				);
+			});
+
+			it("counts both relative and abosolute lost signal counts", () => {
+				sendSignals(60);
+
+				processSubmittedSignals(10);
+
+				dropSignals(1);
+
+				processSubmittedSignals(39);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should log relative lost signal count when a signal is dropped",
+				);
+
+				dropSignals(5);
+
+				sendSignals(45);
+
+				processSubmittedSignals(30);
+
+				dropSignals(4);
+
+				// Process remaining signals
+				processSubmittedSignals(16);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 5,
+						},
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 4,
+						},
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsLost: 10,
```

Comment:
Let's check the send count too. (Just check all of the things that we expect to be true whenever it is reasonable.)

Reply:
There is no sentSignal count on SignalLost events. I thought of including signalsSent, but the concept of what the count should be wasn't intuitive to me. signalsLost here refers to the number of signals we detected have *just been lost* at that moment in time -- but signalsSent could mean the total amount sent up until this point, the amount sent in that current connection window, the amount sent during the roundtrip latency window, etc. 

Could work on this later, but I didn't feel as if signalsSent gave relevant info about the event here.

Reply:
Not suggesting sentCount on Lost event. This is a Latency event and we aren't checking the sentCount but we could (and I say should).

Reply:
RIght I see that, I'll add